ci-operator/step-registry/hypershift/launch/aws/destroy/hypershift-launch-aws-destroy-chain.yaml

```diff
@@ -0,0 +1,45 @@
+chain:
+  as: hypershift-launch-aws-destroy
+  steps:
+  - as: destroy
+    cli: latest
+    env:
+    - name: HYPERSHIFT_BASE_DOMAIN
+      default: "origin-ci-int-aws.dev.rhcloud.com"
+      documentation: "The cluster's FQDN will be a subdomain of the base domain."
+    - name: HYPERSHIFT_AWS_REGION
+      default: "us-east-1"
+      documentation: "The AWS region of the cluster."
+    - name: HYPERSHIFT_NAMESPACE
+      default: "ci-clusters"
+      documentation: "The namespace in which to create hosted clusters."
+    commands: |-
+      set -exuo pipefail
+      export KUBECONFIG=/var/run/hypershift/kubeconfig
+      
+      CLUSTER_NAME="${NAMESPACE}"
+      echo "$(date) Deleting HyperShift cluster ${HYPERSHIFT_NAMESPACE}/${CLUSTER_NAME}"
+      /usr/bin/hypershift destroy cluster \
+        --aws-creds ${CLUSTER_PROFILE_DIR}/.awscred \
+        --namespace "${HYPERSHIFT_NAMESPACE}" \
+        --name ${CLUSTER_NAME} \
+        --region ${HYPERSHIFT_AWS_REGION} \
+        --infra-id "${NAMESPACE}-${CLUSTER_NAME}" \
+        --base-domain ${HYPERSHIFT_BASE_DOMAIN} \
+        --cluster-grace-period 40m
```

Comment:
does this need to be synced with the other grace period, or more accurately, why 40m?

Reply:
Lots to explore in this area...

This flag is _supposed_ to allow infra to be forcibly deleted if the hostedcluster doesn't gracefully delete within a timeout (but upon inspection that code may actually be broken). I actually can't remember why I chose 40, probably because this no longer fails and so I forgot. Let's pretend it works as documented...

If the cluster grace value is less than the job timeout and is exceeded, we probably leak OCP-created infra, and _potentially_ the CLI infra if the latter doesn't complete within the remaining time.

If the cluster grace value is greater than the job timeout and is exceeded, we probably leak the OCP-created infra, and _certainly_ leak the CLI infra.

So maybe a value that fits within the job timeout with room to spare for CLI infra teardown would be advantageous.

Another change we could consider is interrupt handling. Right now a SIGINT during `destroy` will cancel the entire operation. If we treated an INT during hostedcluster teardown as "give up on the hostedcluster but do try and tear down the infra anyway", then the job timeout would apply to the hostedcluster (OCP infra) and the job grace period would apply to the CLI infra teardown part.

On the other hand, these commands should reliably work all the time, in which case these tweaks should be hopefully largely unnecessary. If we do observe instability that would be improved by tweaking these timeouts, we can always adjust this, fix the underlying reliability bugs, etc.

FWIW, so far I have observed our current version to be very stable during my iteration, having made a laundry list of stability improvements to create/destroy that were revealed during this CI exercise.